src/server/game/Entities/Object/Object.h

```diff
@@ -489,8 +490,8 @@ class TC_GAME_API WorldObject : public Object, public WorldLocation
         // use only if you will sure about placing both object at same map
         bool IsWithinDist(WorldObject const* obj, float dist2compare, bool is3D = true) const;
         bool IsWithinDistInMap(WorldObject const* obj, float dist2compare, bool is3D = true) const;
-        bool IsWithinLOS(float x, float y, float z) const;
-        bool IsWithinLOSInMap(WorldObject const* obj) const;
+        bool IsWithinLOS(float x, float y, float z, ObjectIgnoreFlags ignoreFlags = ObjectIgnoreFlags::IGNORE_NONE) const;
```

Comment:
Hmmmm, yea I suppose that would work. Are there guidelines on Enum usage in Trinitycore? My personal preference, since I come from C#, is to include Type information. I think C++11 was moving that direction too with the idea of scoped enums? Not really sure though, my C++ is very limited.

If that's how Trinitycore does enums I'll change it though.